File: 251001/Karpekov/lab4/TaskKafka/src/main/java/lab/controllers/UserController.java
package lab.controllers;

import lab.exceptions.ForbiddenObjectException;
import lab.exceptions.IllegalFieldDataException;
import lab.mappers.UserMapper;
import lab.modelDTOs.UserRequestTo;
import lab.modelDTOs.UserResponseTo;
import lab.models.User;
import lab.services.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.ArrayList;
import java.util.List;

@RestController
@RequestMapping(value = "api/v1.0/users", consumes = {"application/JSON"}, produces = {"application/JSON"})
public class UserController {

    @Autowired
    private UserService userService;

    @GetMapping(value = "")
    ResponseEntity<List<UserResponseTo>> GetUsers(){
        List<UserResponseTo> respList = new ArrayList<>();
        for (User user : userService.getUsers()){
            respList.add(UserMapper.INSTANCE.MapUserToResponseDTO(user));
        }
        return new ResponseEntity<>(respList, HttpStatus.OK);
    }

    @GetMapping(value = "/{id}")
    ResponseEntity<UserResponseTo> GetUserById(@PathVariable("id") long id){
        User u = userService.findById(id);
        return new ResponseEntity<>(UserMapper.INSTANCE.MapUserToResponseDTO(u), HttpStatus.OK);
    }

    @PostMapping(value = "")
    ResponseEntity<UserResponseTo> CreateUser(@RequestBody UserRequestTo newUser){
        User user = UserMapper.INSTANCE.MapRequestDTOToUser(newUser);
        User createUser = userService.createUser(user);
        return new ResponseEntity<>(UserMapper.INSTANCE.MapUserToResponseDTO(createUser), HttpStatus.CREATED);
    }

    @DeleteMapping(value = "/{id}")
    ResponseEntity<UserResponseTo> DeleteUser(@PathVariable("id") long Id){
        userService.deleteUser(Id);
        return new ResponseEntity<>(null, HttpStatus.NO_CONTENT);
    }

    @PutMapping(value = "")
    ResponseEntity<UserResponseTo> UpdateUser(@RequestBody UserRequestTo updUser){
        long Id = updUser.getId();
        User user = UserMapper.INSTANCE.MapRequestDTOToUser(updUser);
        User updateUser = userService.updateUser(user);
        return new ResponseEntity<>(UserMapper.INSTANCE.MapUserToResponseDTO(updateUser), HttpStatus.OK);
    }
}

File: 251003/Kukhatskavolets/microblog/src/main/kotlin/by/kukhatskavolets/controllers/UserController.kt
package by.kukhatskavolets.controllers

import by.kukhatskavolets.dto.requests.UserRequestTo
import by.kukhatskavolets.dto.responses.UserResponseTo
import by.kukhatskavolets.services.UserService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1.0/users")
class UserController(
    private val userService: UserService
) {

    @PostMapping
    fun createUser(@RequestBody @Valid userRequestTo: UserRequestTo): ResponseEntity<UserResponseTo> {
        val createdUser = userService.createUser(userRequestTo)
        return ResponseEntity(createdUser, HttpStatus.CREATED)
    }

    @GetMapping("/{id}")
    fun getUserById(@PathVariable id: Long): ResponseEntity<UserResponseTo> {
        val user = userService.getUserById(id)
        return ResponseEntity.ok(user)
    }

    @GetMapping
    fun getAllUsers(): ResponseEntity<List<UserResponseTo>> {
        val users = userService.getAllUsers()
        return ResponseEntity.ok(users)
    }

    @PutMapping()
    fun updateUser(@RequestBody @Valid userRequestTo: UserRequestTo): ResponseEntity<UserResponseTo> {
        val updatedUser = userService.updateUser(userRequestTo.id, userRequestTo)
        return ResponseEntity.ok(updatedUser)
    }

    @DeleteMapping("/{id}")
    fun deleteUser(@PathVariable id: Long): ResponseEntity<Void> {
        userService.deleteUser(id)
        return ResponseEntity.noContent().build()
    }

    @GetMapping("/by-tweet/{tweetId}")
    fun getUserByTweetId(@PathVariable tweetId: Long): ResponseEntity<UserResponseTo> {
        val user = userService.getUserByTweetId(tweetId)
        return ResponseEntity.ok(user)
    }
}